python/cudf/cudf/tests/test_array_ufunc.py

```diff
@@ -225,19 +221,34 @@ def test_ufunc_series(ufunc, has_nulls, indexed):
 @pytest.mark.parametrize("has_nulls", [True, False])
 @pytest.mark.parametrize("indexed", [True, False])
 @pytest.mark.parametrize("reflect", [True, False])
-def test_binary_ufunc_series_array(ufunc, has_nulls, indexed, reflect):
+def test_binary_ufunc_series_array(
+    request, ufunc, has_nulls, indexed, reflect
+):
     fname = ufunc.__name__
-    if fname in ("greater", "greater_equal", "logical_and") and has_nulls:
-        pytest.xfail(
-            "The way cudf casts nans in arrays to nulls during binops with "
-            "cudf objects is currently incompatible with pandas."
+    request.applymarker(
+        pytest.mark.xfail(
+            condition=reflect and has_nulls,
+            reason=(
+                "When cupy is the left operand there is no way for us to "
+                "avoid calling its binary operators, which cannot handle "
+                "cudf objects that contain nulls."
+            ),
         )
-    if reflect and has_nulls:
-        pytest.skip(
-            "When cupy is the left operand there is no way for us to avoid "
-            "calling its binary operators, which cannot handle cudf objects "
-            "that contain nulls."
+    )
+    # The way cudf casts nans in arrays to nulls during binops with cudf
+    # objects is currently incompatible with pandas.
+    request.applymarker(
+        pytest.mark.xfail(
+            condition=(
+                fname in {"greater", "greater_equal", "logical_and"}
+                and has_nulls
+            ),
+            reason=(
+                "cudf and pandas incompatible casting nans "
+                "to nulls in binops"
+            ),
         )
+    )
```

Comment:
I think the need to applymarker this frequently is an indication that these tests are probably overparametrized. I'm guilty of this since I wrote most of the problematic tests in this file, but it's also a problem that's pretty prevalent throughout the package. Once this PR is merged we may want to follow up with splitting these up.

Reply:
Opened an issue.